### PR TITLE
Feature/allow health ports to be configured

### LINKED ISSUE #508: Amazon EFS Driver Doesn't Install
/kind bug

**What happened?**
Unable to install Amazon EFS driver using with `helm` using [this](https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html) documentation.
Pods named "efs-csi-node" remain in "Pending" status:

```
NAME                                  READY   STATUS    RESTARTS   AGE
efs-csi-controller-59d5dcbfbd-5w5ls   3/3     Running   0          44m
efs-csi-controller-59d5dcbfbd-wgn68   3/3     Running   0          44m
efs-csi-node-4sjpr                    0/3     Pending   0          41m
efs-csi-node-4wdvj                    0/3     Pending   0          41m
efs-csi-node-5ftgm                    0/3     Pending   0          41m
efs-csi-node-6pd5f                    0/3     Pending   0          41m
efs-csi-node-7842g                    0/3     Pending   0          41m
efs-csi-node-8hdb6                    0/3     Pending   0          41m
efs-csi-node-9xl6k                    0/3     Pending   0          41m
efs-csi-node-b4gvq                    0/3     Pending   0          41m
efs-csi-node-b8rs8                    0/3     Pending   0          41m
efs-csi-node-bfb9r                    0/3     Pending   0          41m
efs-csi-node-bqs6p                    0/3     Pending   0          41m
efs-csi-node-dw42m                    0/3     Pending   0          41m
efs-csi-node-f8stm                    0/3     Pending   0          28m
efs-csi-node-fxwtm                    0/3     Pending   0          41m
efs-csi-node-hm66z                    0/3     Pending   0          41m
efs-csi-node-kh97n                    0/3     Pending   0          41m
efs-csi-node-mmk7n                    0/3     Pending   0          41m
efs-csi-node-nq6xj                    0/3     Pending   0          41m
efs-csi-node-pld6m                    0/3     Pending   0          41m
efs-csi-node-qnwx6                    0/3     Pending   0          41m
efs-csi-node-spx8h                    0/3     Pending   0          41m
efs-csi-node-vbvvc                    0/3     Pending   0          41m
efs-csi-node-wb6h9                    0/3     Pending   0          41m
efs-csi-node-wgstj                    0/3     Pending   0          41m
efs-csi-node-wnr8d                    0/3     Pending   0          41m
efs-csi-node-x2f9m                    0/3     Pending   0          41m
efs-csi-node-zqq22                    0/3     Pending   0          41m
```

On describing the one of the "efs-csi-node" pods:

```
Name:                 efs-csi-node-kh97n
Namespace:            adhoc
Priority:             2000001000
Priority Class Name:  system-node-critical
Node:                 <none>
Labels:               app=efs-csi-node
                      app.kubernetes.io/instance=aws-efs-csi-driver
                      app.kubernetes.io/name=aws-efs-csi-driver
                      controller-revision-hash=9d6559fd9
                      pod-template-generation=1
Annotations:          kubernetes.io/psp: eks.privileged
Status:               Pending
IP:
IPs:                  <none>
Controlled By:        DaemonSet/efs-csi-node
Containers:
  efs-plugin:
    Image:      602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/aws-efs-csi-driver:v1.3.2
    Port:       9809/TCP
    Host Port:  9809/TCP
    Args:
      --endpoint=$(CSI_ENDPOINT)
      --logtostderr
      --v=2
    Liveness:  http-get http://:healthz/healthz delay=10s timeout=3s period=2s #success=1 #failure=5
    Environment:
      CSI_ENDPOINT:  unix:/csi/csi.sock
    Mounts:
      /csi from plugin-dir (rw)
      /etc/amazon/efs-legacy from efs-utils-config-legacy (rw)
      /var/amazon/efs from efs-utils-config (rw)
      /var/lib/kubelet from kubelet-dir (rw)
      /var/run/efs from efs-state-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-485r4 (ro)
  csi-driver-registrar:
    Image:      public.ecr.aws/eks-distro/kubernetes-csi/node-driver-registrar:v2.1.0-eks-1-18-2
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=$(ADDRESS)
      --kubelet-registration-path=$(DRIVER_REG_SOCK_PATH)
      --v=2
    Environment:
      ADDRESS:               /csi/csi.sock
      DRIVER_REG_SOCK_PATH:  /var/lib/kubelet/plugins/efs.csi.aws.com/csi.sock
      KUBE_NODE_NAME:         (v1:spec.nodeName)
    Mounts:
      /csi from plugin-dir (rw)
      /registration from registration-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-485r4 (ro)
  liveness-probe:
    Image:      public.ecr.aws/eks-distro/kubernetes-csi/livenessprobe:v2.2.0-eks-1-18-2
    Port:       <none>
    Host Port:  <none>
    Args:
      --csi-address=/csi/csi.sock
      --health-port=9809
      --v=2
    Environment:  <none>
    Mounts:
      /csi from plugin-dir (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-485r4 (ro)
Conditions:
  Type           Status
  PodScheduled   False
Volumes:
  kubelet-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet
    HostPathType:  Directory
  plugin-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/plugins/efs.csi.aws.com/
    HostPathType:  DirectoryOrCreate
  registration-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/kubelet/plugins_registry/
    HostPathType:  Directory
  efs-state-dir:
    Type:          HostPath (bare host directory volume)
    Path:          /var/run/efs
    HostPathType:  DirectoryOrCreate
  efs-utils-config:
    Type:          HostPath (bare host directory volume)
    Path:          /var/amazon/efs
    HostPathType:  DirectoryOrCreate
  efs-utils-config-legacy:
    Type:          HostPath (bare host directory volume)
    Path:          /etc/amazon/efs
    HostPathType:  DirectoryOrCreate
  default-token-485r4:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-485r4
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  beta.kubernetes.io/os=linux
Tolerations:     op=Exists
                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                 node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                 node.kubernetes.io/network-unavailable:NoSchedule op=Exists
                 node.kubernetes.io/not-ready:NoExecute op=Exists
                 node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                 node.kubernetes.io/unreachable:NoExecute op=Exists
                 node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason            Age                   From               Message
  ----     ------            ----                  ----               -------
  Warning  FailedScheduling  2m55s (x62 over 43m)  default-scheduler  0/27 nodes are available: 27 node(s) didn't have free ports for the requested pod ports.

```

**What you expected to happen?**
The pods to be in "Running" status on `kubectl get pod -n adhoc -l "app.kubernetes.io/name=aws-efs-csi-driver,app.kubernetes.io/instance=aws-efs-csi-driver"`

**How to reproduce it (as minimally and precisely as possible)?**
1. Install AWS EFS CSI driver on EKS from [here](https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html)

    a) Create an IAM policy and role

    b) Install the Amazon EFS driver using `helm`:

    ```
    helm repo add aws-efs-csi-driver https://kubernetes-sigs.github.io/aws-efs-csi-driver/

    helm repo update

    helm search repo aws-efs-csi-driver

    helm show values aws-efs-csi-driver/aws-efs-csi-driver > aws-efs-driver.yaml
    ```

    "image.repository" and "controller.serviceAccount.create" values edited in `aws-efs-driver.yaml` because cluster is located in a different region and service account `efs-csi-controller-sa` was created in the previous step.

    Contents of `aws-efs-driver.yaml`:

    ```
    # Default values for aws-efs-csi-driver.
    # This is a YAML-formatted file.
    # Declare variables to be passed into your templates.
    
    nameOverride: ""
    fullnameOverride: ""
    
    replicaCount: 2
    
    image:
      repository: 602401143452.dkr.ecr.us-east-1.amazonaws.com/eks/aws-efs-csi-driver
      tag: "v1.3.2"
      pullPolicy: IfNotPresent
    
    sidecars:
      livenessProbe:
        image:
          repository: public.ecr.aws/eks-distro/kubernetes-csi/livenessprobe
          tag: v2.2.0-eks-1-18-2
          pullPolicy: IfNotPresent
        resources: {}
      nodeDriverRegistrar:
        image:
          repository: public.ecr.aws/eks-distro/kubernetes-csi/node-driver-registrar
          tag: v2.1.0-eks-1-18-2
          pullPolicy: IfNotPresent
        resources: {}
      csiProvisioner:
        image:
          repository: public.ecr.aws/eks-distro/kubernetes-csi/external-provisioner
          tag: v2.1.1-eks-1-18-2
          pullPolicy: IfNotPresent
        resources: {}
    
    imagePullSecrets: []
    
    ## Controller deployment variables
    
    controller:
      # Specifies whether a deployment should be created
      create: true
      # Number for the log level verbosity
      logLevel: 2
      # If set, add pv/pvc metadata to plugin create requests as parameters.
      extraCreateMetadata: true
      # Add additional tags to access points
      tags: {}
        # environment: prod
        # region: us-east-1
      # Enable if you want the controller to also delete the
      # path on efs when deleteing an access point
      deleteAccessPointRootDir: false
      podAnnotations: {}
      resources: {}
        # We usually recommend not to specify default resources and to leave this as a conscious
        # choice for the user. This also increases chances charts run on environments with little
        # resources, such as Minikube. If you do want to specify resources, uncomment the following
        # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
        # limits:
        #   cpu: 100m
        #   memory: 128Mi
        # requests:
        #   cpu: 100m
        #   memory: 128Mi
      nodeSelector: {}
      tolerations: []
      affinity: {}
      # Specifies whether a service account should be created
      serviceAccount:
        create: false
        name: efs-csi-controller-sa
        annotations: {}
        ## Enable if EKS IAM for SA is used
        #  eks.amazonaws.com/role-arn: arn:aws:iam::111122223333:role/efs-csi-role
    
    ## Node daemonset variables
    
    node:
      # Number for the log level verbosity
      logLevel: 2
      hostAliases: {}
        # For cross VPC EFS, you need to poison or overwrite the DNS for the efs volume as per
        # https://docs.aws.amazon.com/efs/latest/ug/efs-different-vpc.html#wt6-efs-utils-step3
        # implementing the suggested solution found here:
        # https://github.com/kubernetes-sigs/aws-efs-csi-driver/issues/240#issuecomment-676849346
        # EFS Vol ID, IP, Region
        # "fs-01234567":
        #   ip: 10.10.2.2
        #   region: us-east-2
      dnsPolicy: ClusterFirst
      dnsConfig: {}
        # Example config which uses the AWS nameservers
        # dnsPolicy: "None"
        # dnsConfig:
        #   nameservers:
        #     - 169.254.169.253
      podAnnotations: {}
      resources: {}
        # limits:
        #   cpu: 100m
        #   memory: 128Mi
        # requests:
        #   cpu: 100m
        #   memory: 128Mi
      nodeSelector: {}
      tolerations:
        - operator: Exists
    
    storageClasses: []
    # Add StorageClass resources like:
    # - name: efs-sc
    #   annotations:
    #     # Use that annotation if you want this to your default storageclass
    #     storageclass.kubernetes.io/is-default-class: "true"
    #   mountOptions:
    #   - tls
    #   parameters:
    #     provisioningMode: efs-ap
    #     fileSystemId: fs-1122aabb
    #     directoryPerms: "700"
    #     gidRangeStart: "1000"
    #     gidRangeEnd: "2000"
    #     basePath: "/dynamic_provisioning"
    #   reclaimPolicy: Delete
    #   volumeBindingMode: Immediate


    ```

    Apply updated values from manifest:

    ```
    helm upgrade --install aws-efs-csi-driver -n adhoc aws-efs-csi-driver/aws-efs-csi-driver --values aws-efs-driver.yaml
    ```

    Verify installation:

    ```
    kubectl get pod -n adhoc -l "app.kubernetes.io/name=aws-efs-csi-driver,app.kubernetes.io/instance=aws-efs-csi-driver"
    ```

**Environment**
- Kubernetes version: 
  + Client: 1.21
  + Server: 1.17
- Driver version: 1.3.2

### PR DESCRIPTION
**Pull request by gezb**
**Is this a bug fix or adding new feature?**

Bug Fix

**What is this PR about? / Why do we need it?**

This PR makes 2 changes 
- Allows the heath ports to be configured (which fixes #508) 
- Adds missing "patch" permission to "events" 

**What testing is done?** 

Rendered the chart out with default values, then modified the new heath port values and re-rendered the chart to prove the new values were used

### CODE REVIEW

File: charts/aws-efs-csi-driver/templates/controller-serviceaccount.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ rules:
     verbs: ["get", "list", "watch"]
   - apiGroups: [""]
     resources: ["events"]
-    verbs: ["list", "watch", "create"]
+    verbs: ["list", "watch", "create", "patch"]
```
Comment (wongma7) [inline charts/aws-efs-csi-driver/templates/controller-serviceaccount.yaml:33]: could you also run `make generate-kustomize` so the kustomize template also picks up this change? thx

Comment (gezb) [reply charts/aws-efs-csi-driver/templates/controller-serviceaccount.yaml:33]: Done

**Comment by wongma7** [comment]
/lgtm

It looks like the helm chart release v2.1.6 didn't go through for some reason, this v2.2.0 might be delayed, I'll figure it out.

**Pull request merged**